heron/proto/stmgr.proto

```diff
@@ -67,4 +67,6 @@ message TupleStreamMessage {
   required int32 task_id = 1;
   // serialized data
   required bytes set = 2;
+  // number of tuples in serialized data
+    required int32 num_tuples = 4;
```

Comment:
need @nwangtw @nlu90 comment on this
if i remember correctly, there was tuple count without this field. how did they count it? @nlu90

Reply:
I think what you are referring is here: https://github.com/apache/incubator-heron/blob/571a90b7c8452f7922cdc47b91e52de66f5edd70/heron/stmgr/src/cpp/manager/instance-server.cpp#L446

Reply:
It is not necessary that every `TupleStreamMessage` be converted into a `HeronTupleSet2` before it is added to the pending queue. In that case, the tuplestreammessage is processed separately (https://github.com/apache/incubator-heron/blob/571a90b7c8452f7922cdc47b91e52de66f5edd70/heron/stmgr/src/cpp/manager/instance-server.cpp#L402) and will not traverse through the code you have linked.